src/bids_validator/__main__.py

```diff
@@ -0,0 +1,56 @@
+import typer
+
+from bids_validator import BIDSValidator
+from bids_validator.types.files import FileTree
+
+validator = BIDSValidator()
+
+app = typer.Typer()
+
+
+def check_children(tree: FileTree):
+    """Iterate over children of a FileTree and check if they are a directory or file.
+
+    If it's a directory then run again recursively, if it's a file file check the file name is
+    BIDS compliant.
+
+    Parameters
+    ----------
+    tree : FileTree
+        FileTree object to iterate over
+
+    """
+    for child in tree.children.values():
+        if child.is_dir:
+            check_children(child)
+        else:
+            check_bids(child.relative_path)
```

Comment:
This is very narrowly usable, but is a pattern we are almost certain to need again. Here is a way to write it to return an iterator:

```python
def walk(tree: FileTree):
    for child in tree.children.values():
        if child.is_dir:
            yield from walk(child)
        else:
            yield child
```

You could then write this check function as:

```python
def validate(tree: FileTree):
    validator = BIDSValidator()
    for file in walk(tree):
        if not validator.is_bids(file.relative_path):
            print(f'{file.relative_path} is not a valid BIDS filename')
```

This separates the concerns of iteration from what to do with the discovered objects.